FreeAPS/Sources/Services/WatchManager/WatchManager.swift

```diff
@@ -288,6 +297,51 @@ extension BaseWatchManager: WCSessionDelegate {
             }
         }
 
+        if let nutrientsData = message["addNutrients"] as? Data {
+            if let nutrients = try? JSONDecoder().decode([Nutrient: Int].self, from: nutrientsData) {
+                if nutrients.values.contains(where: { $0 > 0 }) {
+                    var carbs = min(Decimal(nutrients[.carbs] ?? 0), settingsManager.settings.maxCarbs)
+                    var protein = Decimal(nutrients[.protein] ?? 0)
+                    var fat = Decimal(nutrients[.fat] ?? 0)
+
+                    var useFPUconversion: Bool {
+                        protein > 0 || fat > 0
+                    }
+                    
+                    if useFPUconversion {
+                        let futureCarbArray = calculateFPU(settings: settingsManager, protein: protein, fat: fat)
+                        if !futureCarbArray.isEmpty {
+                            carbsStorage.storeCarbs(futureCarbArray)
+                        }
+                    }
+
+                    // Store the real carbs
+                    carbsStorage.storeCarbs([
+                        CarbsEntry(
+                            id: UUID().uuidString,
+                            createdAt: Date(),
+                            carbs: carbs,
+                            enteredBy: CarbsEntry.manual,
+                            isFPU: false, fpuID: nil
+                        )
+                    ])
```

Comment:
It works the same way as [this method](https://github.com/Artificial-Pancreas/iAPS/blob/dev/FreeAPS/Sources/Modules/AddCarbs/AddCarbsStateModel.swift#L93-L108).